### ISSUE TITLE
Diamond inheritance improperly rendered

### ISSUE THREAD

**Issue by badfd** [doxygen/doxygen#7488]
I searched the repo for this (5-year old) issue, but it seems not to have ever been reported.

Environment:

- Ubuntu 18.04.3
- doxygen/doxywizard 1.8.13-10
- graphviz (dot) 2.40.1-2

Test classes:
```
$ for i in `ls include/*h`; do echo === $i ====================; cat $i; done
=== include/diagrams_a.h ====================
#ifndef _DIAGRAMS_A_H
#define _DIAGRAMS_A_H
class A { public: int _a; };
#endif
=== include/diagrams_b.h ====================
#ifndef _DIAGRAMS_B_H
#define _DIAGRAMS_B_H
#include "diagrams_a.h"
class B : public virtual A { public: int _b; };
#endif
=== include/diagrams_c.h ====================
#ifndef _DIAGRAMS_C_H
#define _DIAGRAMS_C_H
#include "diagrams_a.h"
class C : public virtual A { public: int _c; };
#endif
=== include/diagrams_d.h ====================
#ifndef _DIAGRAMS_D_H
#define _DIAGRAMS_D_H
#include "diagrams_b.h"
#include "diagrams_c.h"
class D : public B, public C { public: int _d; };
#endif
```
When I run `doxywizard` and specify that diagrams be rendered using the "built-in class diagram generator", then the topmost class `A` appears twice in the hierarchy diagram: once as `B`'s base class, and once as `C`'s base class. Of course, having been virtually inherited both by `B` and by `C`, `A` should appear only once. On the plus side, `A` appears in a shaded box with dashed outline.

This dual appearance of `A` (minus shading and dashed outline) would be correct had `A` not been virtually inherited.

If instead of the built-in generator I use GraphViz's `dot` tool, then I get a diamond inheritance hierarchy -- i.e. class `A` appears only once  -- but the box in which `A` appears is not shaded, nor does it have a dashed outline. In other words, `dot` takes no account of the fact that `A` has been virtually inherited.

Related problem: If I change `B` and `C` not to inherit virtually from `A`, then `dot` generates exactly the same diagram, whereas it (arguably) should generate two `A`s. i.e. `dot` is wrong in both cases.

**Comment by albert-github** [comment]
What happens when you use the current doxygen version (1.8.17)?
To be consistent (maybe different doxygen configuration files for each case you described):
- Can you please attach a, small, self contained example (source+configuration file in a tar or zip) that allows us to reproduce the problem?  Please don't add external links as they might not be persistent.

**Comment by badfd** [comment]
> What happens when you use the current doxygen version (1.8.17)?

I hit (first) a `configure` error (test failure), and then a `make` error. I regret that I do not now have to to invetsigate.

> To be consistent (maybe different doxygen configuration files for each case you described):
> 
> * Can you please attach a, small, self contained example (source+configuration file in a tar or zip) that allows us to reproduce the problem?  Please don't add external links as they might not be persistent.

Done. See bottom of OP.
